src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,109 @@
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const emptyMovie: Movie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [count, setCount] = useState(0);
+  const [isError, setError] = useState(true);
+  const [newMovie, setNewMovie] = useState(emptyMovie);
+
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/;
+
+  useEffect(() => {
+    const {
+      title,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = newMovie;
+
+    if (title.trim()
+      && pattern.test(imgUrl)
+      && pattern.test(imdbUrl)
+      && imdbId) {
+      setError(false);
+    } else {
+      setError(true);
+    }
+  }, [newMovie]);
+
+  const onChange = (field: string, value: string) => {
+    setNewMovie((prevMovie) => ({
+      ...prevMovie,
+      [field]: value,
+    }));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd(newMovie);
+    setCount((prevCount) => prevCount + 1);
+
+    setNewMovie(emptyMovie);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newMovie.title}
```

Comment:
```suggestion
        value={title}
```
you can do destructuring here and below